common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookManager.java

```diff
@@ -6,28 +6,44 @@
 
 import com.wynntils.core.WynntilsMod;
 import com.wynntils.core.managers.CoreManager;
+import com.wynntils.mc.utils.ComponentUtils;
+import com.wynntils.mc.utils.ItemUtils;
 import com.wynntils.mc.utils.McUtils;
 import com.wynntils.wynn.event.QuestBookReloadedEvent;
+import com.wynntils.wynn.event.WorldStateEvent;
 import com.wynntils.wynn.model.container.ContainerContent;
 import com.wynntils.wynn.model.container.ScriptedContainerQuery;
 import com.wynntils.wynn.utils.ContainerUtils;
 import com.wynntils.wynn.utils.InventoryUtils;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import net.minecraft.ChatFormatting;
 import net.minecraft.network.chat.TextComponent;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 public class QuestBookManager extends CoreManager {
     private static final int NEXT_PAGE_SLOT = 8;
+    private static final Pattern DIALOGUE_HISTORY_PAGE_PATTERN = Pattern.compile("§7Page \\[(\\d+)/(\\d+)\\]");
 
     private static List<QuestInfo> quests = List.of();
     private static List<QuestInfo> newQuests;
+    private static List<List<String>> dialogueHistory = List.of();
+    private static List<List<String>> newDialogueHistory;
 
     public static void init() {}
 
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    public static void onWorldStateChanged(WorldStateEvent e) {
+        quests = List.of();
```

Comment:
This was a bug that we did not clear quests when changing worlds? :-o

Reply:
Yeah, but we re-queried on screen init so it was okay.